bluepyefe/ecode/sAHP.py

```diff
@@ -109,10 +104,98 @@ def interpret(self, t, current, config_data, reader_data):
         amp2_value = numpy.median(smooth_current[self.tmid : self.tmid2]) - self.hypamp
         self.set_amplitudes_ecode("amp2", config_data, reader_data, amp2_value)
 
-        # Converting back to ms
-        for name_timing in ["ton", "tmid", "tmid2", "toff"]:
-            self.index_to_ms(name_timing, t)
-        self.tend = len(t) * self.dt
+    def step_detection(self, current, config_data, reader_data):
+
+        # Set the threshold to detect the step
+        noise_level = numpy.std(numpy.concatenate((self.current[:50], self.current[-50:])))
+        step_threshold = numpy.clip(4.5 * noise_level, 1e-5, numpy.max(self.current))
+
+        # The buffer prevent miss-detection of the step when artifacts are
+        # present at the very start or very end of the current trace
+        buffer_detect = 2.0
+        idx_buffer = int(buffer_detect / self.dt)
+        idx_buffer = max(1, idx_buffer)
+
+        buffer_step = 50
+        smooth_current = scipy_signal2d(current, 85)
+
+        # Infer the beginning of the long step
+        self.hypamp = base_current(current)
+
+        if "ton" in config_data and config_data["ton"] is not None:
+            self.ton = int(round(config_data["ton"] / self.dt))
+        else:
+            _ = numpy.abs(smooth_current[idx_buffer:] - self.hypamp)
+            self.ton = idx_buffer + numpy.argmax(_ > step_threshold)
+
+        # Infer the end of the long step
+        _ = numpy.flip(
```

Comment:
Same here, I feel like naming variables _ and using them afterwards is not optimal for code reading comprehension